pkg/parse/root_test.go

```diff
@@ -451,6 +457,245 @@ func TestRoot_Parse(t *testing.T) {
 	}
 }
 
+func fakeCRD(opts ...core.MetaMutator) ast.FileObject {
+	crd := fake.CustomResourceDefinitionV1Object(opts...)
+	crd.Spec.Group = "acme.com"
+	crd.Spec.Names = apiextensionsv1.CustomResourceDefinitionNames{
+		Plural:   "anvils",
+		Singular: "anvil",
+		Kind:     "Anvil",
+	}
+	crd.Spec.Scope = apiextensionsv1.NamespaceScoped
+	crd.Spec.Versions = []apiextensionsv1.CustomResourceDefinitionVersion{
+		{
+			Name:    "v1",
+			Served:  true,
+			Storage: false,
+			Schema: &apiextensionsv1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Type: "object",
+					Properties: map[string]apiextensionsv1.JSONSchemaProps{
+						"spec": {
+							Type:     "object",
+							Required: []string{"lbs"},
+							Properties: map[string]apiextensionsv1.JSONSchemaProps{
+								"lbs": {
+									Type:    "integer",
+									Minimum: pointer.Float64Ptr(1.0),
+									Maximum: pointer.Float64Ptr(9000.0),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return fake.FileObject(crd, "cluster/crd.yaml")
+}
+
+func fakeFileObjects() []ast.FileObject {
+	var fileObjects []ast.FileObject
+	for i := 0; i < 500; i++ {
+		fileObjects = append(fileObjects, fake.RoleAtPath(fmt.Sprintf("namespaces/foo/role%v.yaml", i), core.Namespace("foo")))
+	}
+	return fileObjects
+}
+
+func fakeGVKs() []schema.GroupVersionKind {
+	var gvks []schema.GroupVersionKind
+	for i := 0; i < 500; i++ {
+		gvks = append(gvks, schema.GroupVersionKind{Group: fmt.Sprintf("acme%v.com", i), Version: "v1", Kind: "APIService"})
+	}
+	return gvks
+}
+
+func fakeParseError(err error, gvks ...schema.GroupVersionKind) error {
+	groups := make(map[schema.GroupVersion]error)
+	for _, gvk := range gvks {
+		gv := gvk.GroupVersion()
+		groups[gv] = err
+	}
+	return status.APIServerError(&discovery.ErrGroupDiscoveryFailed{Groups: groups}, "API discovery failed")
+}
+
+func TestRoot_Parse_Discovery(t *testing.T) {
+	testCases := []struct {
+		name            string
+		parsed          []ast.FileObject
+		want            []ast.FileObject
+		discoveryClient discoveryutil.ServerResourcer
+		expectedError   error
+	}{
+		{
+			// unknown scoped object should not be skipped when sending to applier when discovery call fails
+			name:            "unknown scoped object with discovery failure of deadline exceeded error",
+			discoveryClient: syncertest.NewDiscoveryClient(context.DeadlineExceeded, kinds.Namespace(), kinds.Role()),
+			expectedError:   fakeParseError(context.DeadlineExceeded, kinds.Namespace(), kinds.Role()),
```

Comment:
Q: Is context.DeadlineExceeded actually the error that gets returned by the real discovery client? It seems like that would be a client-side error, not a server-side error, but I'm not sure. Maybe both use it. Do you know where in the code this is being returned? Maybe add a comment linking to it?

Reply:
I've seem both errors from Spotify's log, and all of them are folded in the same discovery error, then folded in the same APIServerError